src/app/newSettings/PreferencesDialog.tsx

```diff
@@ -53,50 +52,53 @@ const PreferencesDialog = (props: PreferencesDialogProps) => {
     setActiveSection(selectedNavSection);
   };
 
-  const onClose = () => {
+  const onClosePreferences = () => {
     dispatch(uiActions.setIsPreferencesDialogOpen(false));
     navigationService.closePreferencesDialog();
   };
 
   return (
     <Modal
       maxWidth="max-w-4xl"
-      className="dark:dbg-surface m-0 flex h-640 shadow-sm"
+      className="m-0 flex h-640 overflow-hidden shadow-sm"
       isOpen={isPreferencesDialogOpen}
-      onClose={() => onClose()}
+      onClose={() => onClosePreferences()}
+      // preventClosing={isToastNotificationOpen}
     >
       <Helmet>
         <title>{title} - Internxt Drive</title>
       </Helmet>
-
       {/* SIDEBAR MENU */}
-      <section className="w-56 border-r border-gray-10 px-2.5">
+      <section className="w-56 shrink-0 border-r border-gray-10 px-2.5">
         <h1 className="py-3 pl-4 text-xl font-semibold">{translate('preferences.title')}</h1>
         <SectionListContainer activeSection={activeSection} changeSection={changeSection} />
       </section>
       {/* SIDEBAR MENU */}
-
-      {/* ACTIVE SECTION */}
-      <section className="relative w-full overflow-y-auto">
-        <button className="fixed right-0 z-50 m-4 flex w-auto" onClick={() => onClose()}>
-          <X size={22} />
-        </button>
-        {activeSection?.section === 'general' && activeSection?.subsection === 'general' && <GeneralSection />}
-        {activeSection?.section === 'workspace' && activeSection?.subsection === 'overview' && <OverviewSection />}
-        {activeSection?.section === 'workspace' && activeSection?.subsection === 'members' && <MembersSection />}
+      <section className="w-full">
+        {activeSection?.section === 'general' && activeSection?.subsection === 'general' && (
+          <GeneralSection onClosePreferences={() => onClosePreferences()} />
+        )}
+        {activeSection?.section === 'workspace' && activeSection?.subsection === 'overview' && (
+          <OverviewSection onClosePreferences={() => onClosePreferences()} />
+        )}
+        {activeSection?.section === 'workspace' && activeSection?.subsection === 'members' && (
+          <MembersSection onClosePreferences={() => onClosePreferences()} />
+        )}
         {activeSection?.section === 'workspace' && activeSection?.subsection === 'billing' && (
-          <BillingWorkspaceSection />
+          <BillingWorkspaceSection onClosePreferences={() => onClosePreferences()} />
```

Comment:
no need to add in a anonymous function, could pass the reference directly. That way is better for the performance, otherwise the anonymous function is destroyed and created for each render

Reply:
Done